lambda/src/main/resources/db/migration/V15__add-consignment-seq-and-reference.sql

```diff
@@ -0,0 +1,15 @@
+-- Create sequence that will be the consignment sequence ID used to encode b25 section of consignment reference
+CREATE SEQUENCE "ConsignmentSequenceID"
+NO CYCLE
+START WITH 1
+INCREMENT BY 1;
+
+-- Alter Consignment table to add column to store sequence ID and another for full consignment reference
+ALTER TABLE "Consignment"
+ADD COLUMN "ConsignmentSequence" bigint DEFAULT nextval('"ConsignmentSequenceID"'),
+ADD COLUMN "ConsignmentReference" text,
+ADD UNIQUE ("ConsignmentSequence", "ConsignmentReference");
```

Comment:
A couple of things about the uniqueness constraint:

The columns should probably be made separately unique. This constraint means "the combination of ConsignmentSequence and ConsignmentReference must be unique", but it technically means there could be duplicate consignment references. That's unlikely to happen in practice because the code saves a reference based on the sequence column, but the database constraints are there to give us some extra guarantees in case the code does the wrong thing.

I'm not totally sure about this, but I think it's better to use `ADD CONSTRAINT constraintname UNIQUE (columns)` rather than `ADD UNIQUE (columns)`. The latter generates a constraint name automatically, but it's helpful to name the constraint because it makes it easier to delete or change it in future if necessary.

Reply:
Ah I didn't know that about the `UNIQUE` constraints! I'll change it so they're separately unique :+1:

Reply:
It looks like the uniqueness constraint still groups the columns?

```
ADD CONSTRAINT constraint_name UNIQUE ("ConsignmentSequence", "ConsignmentReference");
```

I think you'll have to add them separately.

Also, what does `constraint_name` refer to? I'd have expected a string like "UniqueConsignmentReference".

Reply:
Looks like this constraint still needs to be split up.